### ISSUE TITLE
rpm packages are not building after 0.150.0

### ISSUE THREAD

**Issue by hellt** [goreleaser/goreleaser#2033]
**Describe the bug**

NFPM doesn't include the binary to the rpm package, wheres it does for deb. I noticed this changed since `0.150`. I.e. `0.150.0` still works, but releases after it (0.152.0 for example) don't.

**To Reproduce**

```yaml
nfpms:
  - id: test
    file_name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}"
    package_name: test
    license: GNU GPLv3
    formats:
      - rpm
      - deb
```

The resulting content of `dist`:
```
❯ ls -lah dist
total 32M
drwxr-xr-x  4 root root 4.0K Jan 25 17:08 .
drwxr-xr-x 10 root root  308 Jan 25 17:08 ..
-rw-r--r--  1 root root  854 Jan 25 17:08 checksums.txt
-rw-r--r--  1 root root 2.5K Jan 25 17:08 config.yaml
drwxr-xr-x  2 root root   26 Jan 25 17:08 containerlab_linux_386
drwxr-xr-x  2 root root   26 Jan 25 17:08 containerlab_linux_amd64
-rw-r--r--  1 root root 5.2M Jan 25 17:08 containerlab_v0.9.0-next_linux_386.apk
-rw-r--r--  1 root root 5.2M Jan 25 17:08 containerlab_v0.9.0-next_linux_386.deb
-rw-r--r--  1 root root  33K Jan 25 17:08 containerlab_v0.9.0-next_linux_386.rpm
-rw-r--r--  1 root root 5.1M Jan 25 17:08 containerlab_v0.9.0-next_Linux_386.tar.gz
-rw-r--r--  1 root root 5.4M Jan 25 17:08 containerlab_v0.9.0-next_linux_amd64.apk
-rw-r--r--  1 root root 5.4M Jan 25 17:08 containerlab_v0.9.0-next_linux_amd64.deb
-rw-r--r--  1 root root  33K Jan 25 17:08 containerlab_v0.9.0-next_linux_amd64.rpm
-rw-r--r--  1 root root 5.4M Jan 25 17:08 containerlab_v0.9.0-next_Linux_amd64.tar.gz
```

**Expected behavior**

rpm artifacts contain binary

**Environment (please complete the following information):**

- OS: mac

**Comment by caarlos0** [comment]
Just ran:

```sh
$ docker run -it --rm fedora
# yum install https://github.com/goreleaser/goreleaser/releases/download/v0.155.0/goreleaser_amd64.rpm
# ommited lines

# goreleaser -v
goreleaser version 0.155.0
commit: ba82f43c5f2eefcc82d8d14634fe21b37ffc1799
built at: 2021-01-15T12:54:31Z
built by: goreleaser
```

Not sure why yours is not working 🤔  can you share a reproducible?

**Comment by hellt** [comment]
@caarlos0 I am sorry, maybe I was not clear enough. The issue is not with rpm package of the goreleaser, but with goreleaser not building rpms with nfpm integration.

**Comment by caarlos0** [comment]
I got it, yes, but goreleaser itself is released with goreleaser, and the rpm is as expected there...

**Comment by hellt** [comment]
I see, thank you. Maybe it's something on my end, some mix of options that
didn't work out with 0.150.0+, but I won't have time to shuffle then to see
what's going on, unless there is a debug mode that can pinpoint the
problem.

Maybe worth noting that symlinks didn't work either, as well as wildcards
with `content` paths

**Comment by hellt** [comment]
Fwiw, here is the  config file where things didn't work out

https://github.com/srl-wim/container-lab/blob/master/.goreleaser.yml

Here I reverted it to 0.149.0 compatible version for nfpm files and symlinks, but I changed them accordingly when tried 0.155

**Comment by caarlos0** [comment]
you can run with `--debug` which should yield more logs... those would help debug the issue for sure

**Comment by hellt** [comment]
nothing suspicious in the logs.
For the sake of completeness, here is the config.yaml that is generated by goreleaser and put into `dist`:

https://gist.github.com/hellt/759076514f9f75e654456227015eafb2

**Comment by caarlos0** [comment]
Maybe the symlink is the issue?

```yaml
-  src: /usr/bin/clab
    dst: /usr/bin/containerlab
    type: symlink
```

I think src and dst should be the other way around, no? 🤔

**Comment by hellt** [comment]
thanks @caarlos0 , that indeed was it
The interesting fact is that `goreleaser` made this translation from the old nfpm syntax to a new one that you pasted. The old one works as it appears in the gist, so I just followed the same pattern with a raised eyebrow (why src is dst and dst is src). But I dared not to try and swap the src and dst, since that was the way it worked in 0.149.0

also funny, that only rpm packaging system showed that something was wrong.

Now it seems all good, so I am closing this issue
